lib/mailer/index.js

```
@@ -32,48 +33,14 @@ if (!config.active) {
   console.warn('Emails will not be sent!!')
 }
 
-const client = new mandrill.Mandrill(config.mailchimpKey)
+let provider
 
-client.sendEmail = function (conf) {
-  var q = new Promise(function (resolve, reject) {
-    var message = {
-      from_email: conf.sender.email,
-      from_name: conf.sender.name
-    }
-
-    if (!config.active) {
-      console.log(`Email not send, body => \n ${conf.body} \n`)
-      return resolve()
-    }
-
-    message.html = conf.body
-    message.subject = conf.title
-    if (conf.recipient) {
-      message.to = [{
-        email: conf.recipient.email,
-        name: conf.recipient.name
-      }]
-    } else {
-      message.to = conf.recipients
-    }
-
-    if (!config.active) {
-      console.log(`Email not send, body => \n ${conf.body} \n`)
-      return resolve()
-    }
-
-    client.messages.send({
-      'message': message,
-      'async': false,
-      'ip_pool': 'Main pool'
-    }, function (result) {
-      resolve(result)
-    }, function (err) {
-      reject(err)
-    })
-  })
-
-  return q
+if (config.emailProvider === 'mandrill') {
+  provider = new Mandrill(config)
+} else if (config.emailProvider === 'sendgrid') {
+  provider = new SendGrid(config)
+} else {
+  console.log('Email provider not found. You can define it on your env file.')
```

this should throw an error and not only a console log

---

Maybe we can verify:
```
if (!active) {
  console.log(body)
} else if (active && !emailProviders.hasOwnProperty(config.emailProvider)) {
  throw Exception("Email provider isn't configured!")
} else {
  ...
}
```

---

And remove the lines:
```
if (!config.active) {
        console.log(`Email not send, body => \n ${conf.body} \n`)
        return resolve()
}
```

---

I think this is ok with resolve, but if there is no a valid provider maybe `mailer.send` should throw an error... because we can be able to not specifying a provider but make sure that there is a good or valid one

---

I mean, this exception should be thrown only if we're attempting to send an email with an invalid provider `throw Exception("Email provider isn't configured!")` into `mailer.send` method